wp-content/plugins/core/src/Blocks/Types/Hero/Model.php

```diff
@@ -0,0 +1,134 @@
+<?php
```

Comment:
Not sure I follow the purpose for this model or at least in the core plugin? This seems more hero block controller'esque based on what we've been doing thus far, namely with all the setting of classes & markup.

Reply:
Yeah. I thought the same when Mike first showed me this arch model today. However, I was brought around to this idea b/c it allows our block "components" to stay data-source agnostic. Meaning, unlike a template component or dumb component, "blocks" need to source their data from somewhere. And we want to keep the block component agnostic of that.

It allows the blocks to be used as a "true" component that doesn't care about the data source. In this way,  the hero block _component_ could power both a page's subheader via meta-fields on the post type _and_ a hero block within the page's content.

Reply:
Ok, that's helpful, however, that also sounds like what we were moving away from in this pivot. We still have that capability for our non route / content "component"s and can keep it in the theme, which reduces cognitive overhead. For the sake of consistency, I'd vote this goes in the theme like we've done with all the other templates/components now. My two cents. I'll let @jbrinley and @nickpelton weigh in and also vote. Regardless, thanks for knocking out the first pass, Mike ❤️